src/Controller.js

```diff
@@ -342,6 +343,21 @@ class Controller {
 		this.$link = this.getButton().find( 'a' );
 		return this.$link;
 	}
+
+	/**
+	 * Get the overlay div with the class wwt-overlay
+	 * Will add it if it doesn't already exist in the DOM.
+	 *
+	 * @return {jQuery} Overlay div
+	 */
+	getOverlay() {
+		if ( this.$overlay instanceof $ ) {
```

Comment:
I'm not a huge fan of 'get' methods that also make the widget/element, I prefer we create the element once on init and then attach it, and have the get method return the element we know exists already.

That said, I know that we do this for the getButton so while I am not a huge fan, I think consistency is okay here. I do think we might want to take a look in the future to make sure this doesn't create some annoyances like creating/attaching an overlay when it's not needed (like happened with the button), so maybe just keep our eyes open for a potential future bug.

Reply:
Interesting. I actually feel the opposite, I think! :) It's also the same pattern that `OO.ui.getDefaultOverlay()` uses. It sort of makes sense to me, in that it ensures we only create and attach the thing when we know we need it (and we know we need it because someone called `getOverlay()`; no extra context required).